units/host_monitoring_check.go

```diff
@@ -92,10 +89,48 @@ func (j *hostMonitorExternalStateCheckJob) Run(ctx context.Context) {
 		j.env = evergreen.GetEnvironment()
 	}
 
+	if j.host.Provider == evergreen.ProviderNameStatic {
+		j.AddError(j.handleUnresponsiveStaticHost(ctx))
+		return
+	}
+
 	_, err = handleExternallyTerminatedHost(ctx, j.ID(), j.env, j.host)
 	j.AddError(err)
 }
 
+// handleUnresponsiveStaticHost checks if a static host has been unresponsive
+// for a long time and if so, quarantines it. This is an imprecise check,
+// because static hosts do not have a reliable way to verify if they're actually
+// healthy and able to run tasks. That means that in some rare cases, this could
+// produce false positives (e.g. if Evergreen is down for hours, the static
+// hosts can't reach it) and false negatives (because it waits a very long time
+// before declaring a host unhealthy).
+func (j *hostMonitorExternalStateCheckJob) handleUnresponsiveStaticHost(ctx context.Context) error {
+	if j.host.Provider != evergreen.ProviderNameStatic {
+		return nil
+	}
+	if j.host.Status == evergreen.HostQuarantined {
```

Comment:
can we instead check if status is not in `evergreen.DownHostStatus`?

Reply:
Hmm yeah that works as well. Switched it to that.